### PR TITLE
fix all ts warnings in arb-provider-ethers

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

File: packages/arb-provider-ethers/src/lib/value.ts [hunk 1]
```diff
@@ -612,11 +609,6 @@ function _unmarshalValue(array: Uint8Array, offset: number): [Value, number] {
     const nextHash = ethers.utils.hexlify(head)
     return [new CodePointValue(pc, op, nextHash), offset]
   } else if (ty === ValueType.HashOnly) {
-    ;[head, offset] = extractBytes(array, offset, 8)
-    const size = ethers.utils.bigNumberify(head)
-    ;[head, offset] = extractBytes(array, offset, 32)
-    const hash = '0x' + head
-    // return [new HashOnlyValue(hash, size), tail];
     throw Error('Error unmarshaling: HashOnlyValue was not expected')
```
Comment (DZGoldman) [inline packages/arb-provider-ethers/src/lib/value.ts:612]: sanity check here — is this the intent here indeed to always throw an error for this condition?

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5782094&bid=18930831)

File: packages/arb-provider-ethers/src/lib/value.ts (hunk 1 above)
Comment (hkalodner) [reply packages/arb-provider-ethers/src/lib/value.ts:612]: Yep 😁

File: .eslintrc.js [hunk 2]
```diff
@@ -37,10 +37,10 @@ module.exports = {
         ],
         '@typescript-eslint/no-use-before-define': [
           'warn',
-          { functions: true },
+          { functions: false },
         ],
         '@typescript-eslint/member-delimiter-style': ['off'],
-        '@typescript-eslint/no-explicit-any': ['warn'],
+        '@typescript-eslint/no-explicit-any': ['off'],
```
[the remark this answers is not in the document]
Comment (DZGoldman) [reply .eslintrc.js:43]: oh gotcha, misunderstood what you said; disable case by case w/ `// @ts-ignore` i assume

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5782094&bid=18930831)

**Comment by hkalodner** [comment]
I swapped this over to being based on develop and merged up to latest

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5782094&bid=18930831)

**Pull request merged**